crates/turborepo-analytics/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+#![deny(clippy::all)]
+
+use std::time::Duration;
+
+use futures::stream::FuturesUnordered;
+use thiserror::Error;
+use tokio::{
+    select,
+    sync::{mpsc, oneshot},
+    task::{JoinError, JoinHandle},
+};
+use tracing::debug;
+use turborepo_api_client::{APIAuth, APIClient};
+pub use turborepo_vercel_api::AnalyticsEvent;
+use uuid::Uuid;
+
+const BUFFER_THRESHOLD: usize = 10;
+
+static EVENT_TIMEOUT: Duration = Duration::from_millis(200);
+static NO_TIMEOUT: Duration = Duration::from_secs(24 * 60 * 60);
+static REQUEST_TIMEOUT: Duration = Duration::from_secs(10);
+const CHANNEL_SIZE: usize = 100;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to send analytics event")]
+    SendError(#[from] mpsc::error::SendError<AnalyticsEvent>),
+    #[error("Failed to record analytics")]
+    Join(#[from] JoinError),
+}
+
+// We have two different types because the AnalyticsSender should be shared
+// across threads (i.e. Clone + Send), while the AnalyticsHandle cannot be
+// shared since it contains the structs necessary to shut down the worker.
+pub type AnalyticsSender = mpsc::Sender<AnalyticsEvent>;
+
+pub struct AnalyticsHandle {
+    exit_ch: oneshot::Receiver<()>,
+    handle: JoinHandle<()>,
+}
+
+pub fn start_analytics(api_auth: APIAuth, client: APIClient) -> (AnalyticsSender, AnalyticsHandle) {
+    let (tx, rx) = mpsc::channel(CHANNEL_SIZE);
```

Comment:
Do we want to use an unbounded channel here? We don't necessarily want back pressure from sending analytic events. A cache operation should never be blocked from completing if the analytics channel is full. (I realize this is unlikely to ever happen)